File: backend-java-reactive/src/main/java/com/beautysalon/reactive/model/Service.java
package com.beautysalon.reactive.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.cassandra.core.mapping.PrimaryKey;
import org.springframework.data.cassandra.core.mapping.Table;

import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Table("services")
public record Service(
    @PrimaryKey @Id UUID id,
    @NotBlank(message = "Name is required") String name,
    String description,
    @NotNull(message = "Price is required")
    @DecimalMin(value = "0.0", inclusive = false, message = "Price must be greater than 0")
    BigDecimal price,
    @NotNull(message = "Duration is required")
    @Min(value = 1, message = "Duration must be at least 1 minute")
    Integer durationMinutes,
    @NotBlank(message = "Category is required") String category,
    Boolean active,
    LocalDateTime createdAt,
    LocalDateTime updatedAt
) {
    public Service {
        if (id == null) {
            id = UUID.randomUUID();
        }
        if (active == null) {
            active = true;
        }
        if (createdAt == null) {
            createdAt = LocalDateTime.now();
        }
        if (updatedAt == null) {
            updatedAt = LocalDateTime.now();
        }
    }

    public static Service create(String name, String description, BigDecimal price, 
                               Integer durationMinutes, String category) {
        return new Service(null, name, description, price, durationMinutes, category, null, null, null);
    }

    public Service withUpdatedFields(String name, String description, BigDecimal price, 
                                   Integer durationMinutes, String category, Boolean active) {
        return new Service(
            this.id,
            name != null ? name : this.name,
            description != null ? description : this.description,
            price != null ? price : this.price,
            durationMinutes != null ? durationMinutes : this.durationMinutes,
            category != null ? category : this.category,
            active != null ? active : this.active,
            this.createdAt,
            LocalDateTime.now()
        );
    }
}

File: backend-dotnet/BeautySalonAPI/Models/Service.cs
using System.ComponentModel.DataAnnotations;

namespace BeautySalonAPI.Models;

public class Service
{
    [Key]
    public Guid Id { get; set; }

    [Required(ErrorMessage = "Name is required")]
    [StringLength(100, ErrorMessage = "Name cannot exceed 100 characters")]
    public string Name { get; set; } = string.Empty;

    [StringLength(500, ErrorMessage = "Description cannot exceed 500 characters")]
    public string? Description { get; set; }

    [Required(ErrorMessage = "Price is required")]
    [Range(0.01, double.MaxValue, ErrorMessage = "Price must be greater than 0")]
    public decimal Price { get; set; }

    [Required(ErrorMessage = "Duration is required")]
    [Range(1, int.MaxValue, ErrorMessage = "Duration must be at least 1 minute")]
    public int DurationMinutes { get; set; }

    [Required(ErrorMessage = "Category is required")]
    [StringLength(50, ErrorMessage = "Category cannot exceed 50 characters")]
    public string Category { get; set; } = string.Empty;

    public bool Active { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public Service()
    {
        Id = Guid.NewGuid();
        Active = true;
        CreatedAt = DateTime.UtcNow;
        UpdatedAt = DateTime.UtcNow;
    }

    public static Service Create(string name, string? description, decimal price, int durationMinutes, string category)
    {
        return new Service
        {
            Name = name,
            Description = description,
            Price = price,
            DurationMinutes = durationMinutes,
            Category = category
        };
    }

    public Service Update(string? name = null, string? description = null, decimal? price = null,
                         int? durationMinutes = null, string? category = null, bool? active = null)
    {
        return new Service
        {
            Id = this.Id,
            Name = name ?? this.Name,
            Description = description ?? this.Description,
            Price = price ?? this.Price,
            DurationMinutes = durationMinutes ?? this.DurationMinutes,
            Category = category ?? this.Category,
            Active = active ?? this.Active,
            CreatedAt = this.CreatedAt,
            UpdatedAt = DateTime.UtcNow
        };
    }
}
